safe-global/safe-wallet-web
# [Seedless Onboarding]: Add password recovery modal design

## What it solves

Part of #2452

## How this PR fixes it

- Implements designs for the password recovery prompt
- Reuses `setTxFlow` for the modal
- Shows an error if the submitted password is wrong

## How to test it

1. Open Safe on a new device
2. Log in via Google from the Header
3. Observe the new password recovery modal design
4. Type in a wrong password and submit
5. Observe an error
6. Type in the correct password
7. Observe getting logged in
8. Log out and navigate to the welcome page
9. Press Continue with Google
10. Do the same thing
11. Observe being navigated to the safe creation flow after

## Screenshots
<img width="760" alt="Screenshot 2023-10-18 at 18 19 01" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/c823fea7-f354-4bf7-9fb4-9ff400d6a587">
<img width="653" alt="Screenshot 2023-10-18 at 18 19 47" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/42c103bf-4fcb-4955-935a-425e296c4103">
<img width="584" alt="Screenshot 2023-10-20 at 10 52 13" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/fd57bfd5-4608-4c2e-8766-8e087f2024de">


## Checklist
